inflate_p.h

```diff
@@ -146,7 +148,7 @@ static inline uint8_t* chunkcopy_safe(uint8_t *out, uint8_t *from, size_t len, u
      * initial bulk memcpy of the nonoverlapping region. Then, we can leverage the size of this to determine the safest
      * atomic memcpy size we can pick such that we have non-overlapping regions. This effectively becomes a safe look
      * behind or lookahead distance */
-    size_t non_olap_size = ((from > out) ? from - out : out - from);
+    size_t non_olap_size = labs(from - out);
```

Comment:
I'm not seeing where labs is defined.

Reply:
It's in math.h I think, it's just the long version of abs

Reply:
In my Linux, it's in stdlib.h.

Reply:
> it's just the long version of abs

Ah, it the name just looks a little weird that is why I didn't recognize it. 😁

Reply:
This is a wrong change.

The ``from`` and ``out`` are pointers, ``typeof(from - out)`` is ``ptrdiff_t``.

```c
long labs( long n );
```

But, on Win64:
``long`` - 32 bit
pointers, ``size_t`` and ``ptrdiff_t`` - 64 bit

See: https://pvs-studio.com/en/blog/terms/0012/

Reply:
What is the 64 bit abs on Windows if not labs?  Bear in mind we're doing this to get around some string fortification that GCC is unwisely flagging as potentially hazardous when we were doing this math as a ternary statement (it detected an unrealistic bounds from the pointer arithmetic that in practice never happened because the condition of comparing the greater pointer prior to the subtraction meant that the size never went negative).

Reply:
Ah, a little bit of google sleuthing seems to reveal llabs is what we want?  I never understood Microsoft's hack that was ILP32LL - it just seems like they could have written most if not all of their code to used fixed width types if they needed this compatibility this badly.

Reply:
``long long`` is always 64 bit (on all 32 and 64 bit platforms).
```
long long llabs( long long n ); (since C99)
Defined in header <inttypes.h>
```
(https://en.cppreference.com/w/c/numeric/math/abs)